engine/action_contact.go

```diff
@@ -163,6 +163,78 @@ func (c *Client) UpdateContact(ctx context.Context, id, fullName string, metadat
 	return contact, nil
 }
 
+// AdminCreateContact creates a new contact - xpubId is retrieved by the creatorPaymail.
+func (c *Client) AdminCreateContact(ctx context.Context, contactPaymail, creatorPaymail, fullName string, metadata *Metadata) (*Contact, error) {
+	if err := validateNewContactReqFields(fullName, creatorPaymail); err != nil {
+		return nil, err
+	}
+
+	creatorPaymailAddr, err := getPaymailAddress(ctx, creatorPaymail, c.DefaultModelOptions()...)
+	if err != nil {
+		return nil, spverrors.ErrCouldNotFindPaymail.Wrap(err)
+	}
+	if creatorPaymailAddr == nil {
+		return nil, spverrors.ErrCouldNotFindPaymail
+	}
+
+	creatorXPub, err := getXpubByID(ctx, creatorPaymailAddr.XpubID, c.DefaultModelOptions()...)
+	if err != nil {
+		return nil, spverrors.ErrCouldNotFindXpub.Wrap(err)
+	}
+
+	newContactSanitisedPaymail, err := c.PaymailService().GetSanitizedPaymail(contactPaymail)
+	if err != nil {
+		return nil, spverrors.Wrapf(err, "requested duplicate paymail is invalid")
+	}
+
+	pkiNewContact, err := c.PaymailService().GetPkiForPaymail(ctx, newContactSanitisedPaymail)
+	if err != nil {
+		return nil, spverrors.ErrGettingPKIFailed.Wrap(err)
+	}
+
+	duplicate, err := getContact(ctx, contactPaymail, creatorXPub.ID, c.DefaultModelOptions()...)
+	if err != nil {
+		return nil, err
+	}
+	if duplicate != nil {
+		return nil, spverrors.ErrContactAlreadyExists
+	}
+
+	opts := c.DefaultModelOptions()
+	if metadata != nil {
+		for key, value := range *metadata {
+			opts = append(opts, WithMetadata(key, value))
+		}
+	}
+
+	contact := newContact(
+		fullName,
+		contactPaymail,
+		pkiNewContact.PubKey,
+		creatorXPub.ID,
+		// newly created contact should be in the status of ContactNotConfirmed - initial state
+		ContactNotConfirmed,
```

Comment:
I'm not sure if there shouldn't be an `awaiting` status here

Reply:
I would keep not confirmed as we are not waiting for anything - anything was designed if we invite someone and we are waiting if he accepts or rejects invitation.